Modules/Registration/Metricsv4/include/itkMattesMutualInformationImageToImageMetricv4.hxx

```diff
@@ -366,7 +388,10 @@ MattesMutualInformationImageToImageMetricv4<TFixedImage,
   PDFValueType sum = 0.0;
   for (unsigned int fixedIndex = 0; fixedIndex < temp_num_histogram_bins; ++fixedIndex)
   {
-    const PDFValueType fixedImageMarginalPDFValue = this->m_ThreaderFixedImageMarginalPDF[0][fixedIndex];
+    const PDFValueType fixedImageMarginalPDFValue = l_FixedImageMarginalPDF[fixedIndex];
+
+    // const bool         isNotNearZerofixedImageMarginalPDFValue = (fixedImageMarginalPDFValue > closeToZero);
+    // if (isNotNearZerofixedImageMarginalPDFValue)
```

Comment:
Is this commented out on purpose?

Reply:
I now see that a subsequent commit refactors this.

Reply:
@dzenanz Actually I forgot to make the most important last commit that makes the actual fix!!.

I know this is not a perfect set of patches, but I have reworked this a few times trying to be very methodical in the changes to such a critical piece of code.